src/txn/txn_ckpt.c

```diff
@@ -1519,7 +1536,15 @@ __wt_txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[], bool waiting)
         WT_WITH_CHECKPOINT_LOCK(session, ret = __txn_checkpoint_wrapper(session, cfg));
     else
         WT_WITH_CHECKPOINT_LOCK_NOWAIT(session, ret, ret = __txn_checkpoint_wrapper(session, cfg));
+    /*
+     * If this checkpoint is flushing objects, a failure can leave a tree's block manager pointing
+     * to incorrect blocks. Currently we can not recover from this situation. Panic!
```

Comment:
I'm OK with this failure mode. 

Do you (or @ddanderson) think that this is a condition we should try to handle in the future? I.e., is it realistic to hit it and feasible to handle it? If so, we should have a ticket.

Reply:
Filed [WT-12052](https://jira.mongodb.org/browse/WT-12052). I talked to @agorrod about the current error handling for checkpoint. Early on in the checkpoint process failures could be ignored by the application, but the expectation in general is that checkpoint failures are non-recoverable, and the system must panic. So we are safe for now. But the team can talk more in [WT-12052](https://jira.mongodb.org/browse/WT-12052).

Reply:
I had a brief chat with Alex about the current expectations with checkpoint failures. I think it is worth investigating what is the point of no return for the current checkpoint code. Certainly, early in the checkpoint call where we are sanitizing the provided configuration, errors are safely handled. I am not sure what happens if there is a failure in the prepare stage - we have certainly not started to make changes that need to be reversed. So without flush tier if a prepare failure is handled gracefully - with tier flush that will not be the case. Prepare stage schedules a switch of the file and pushes a work object in the tiered server queue - so those things would need to be undone. If prepare failure without flush tier is considered fatal - we can just continue to do that with flush tier as well.

Reply:
Thanks for the ticket and the followup conversation with Alex.  I'm good with this now.